### PR TITLE
[tests] Proxy/XXXRequest

### PR DESCRIPTION
**Pull request by alagroy-42**
Here come the unit tests for the `Lizmap\Request` namespace classes. There is still some methods untested particualrly the ones that are retrieving data from an external service or the one that are handling a big volume of data that I don't have access to (XML/GML etc...) If you send me samples from these data (some valid and some invalid if it's possible), I will be able to write these tests.

### CODE REVIEW

File: tests/units/classes/Request/ClassesForTests.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        return $this->loginFilters;
+    }
+
+    public function setData($key, $value)
+    {
+        $this->data[$key] = $value;
+    }
+}
+
+class jTpl
```
Comment (laurentj) [inline tests/units/classes/Request/ClassesForTests.php:28]: You cannot redefine classes that already exists, like jTpl or lizmapTiler, else we couldn't test components which are using them and need real results from them.And we couldn't test these classes too (at least lizmapTiler).

**Review by laurentj — CHANGES_REQUESTED** [review]

File: lizmap/modules/lizmap/lib/Request/WMSRequest.php [hunk 2]
```diff
@@ -188,7 +187,8 @@ protected function getcontext()
             array('repository' => $this->repository->getKey(), 'project' => $this->project->getKey())
         );
         $sUrl = str_replace('&', '&amp;', $sUrl).'&amp;';
-        $data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $response->data);
+        $data = $response->data;
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/Request/WMSRequest.php:190]: why this change?

**Review by laurentj — APPROVED** [review]

File: lizmap/modules/lizmap/lib/Request/WMSRequest.php (hunk 2 above)
Comment (alagroy-42) [reply lizmap/modules/lizmap/lib/Request/WMSRequest.php:190]: https://github.com/3liz/lizmap-web-client/pull/1943#pullrequestreview-541093390
